app/v2/controllers/V2MovementsController.scala

```diff
@@ -314,6 +319,34 @@ class V2MovementsControllerImpl @Inject() (
         }
     }
 
+  private def acceptedMessageBody(
+    messageSummary: MessageSummary,
+    acceptHeader: String,
+    bodyAndSizeMaybe: Option[BodyAndSize]
+  )(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, PresentationError, BodyAndContentType] = {
+    def bodyExists(bodyAndSize: BodyAndSize): EitherT[Future, PresentationError, BodyAndContentType] =
+      acceptHeader match {
+        case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON if bodyAndSize.size > config.smallMessageSizeLimit =>
+          EitherT.leftT[Future, BodyAndContentType](PresentationError.notAcceptableError("Large messages cannot be returned as json"))
```

Comment:
I think we probably need a better error message than this - we might want to put the size limit or something like that. Might be worth talking to Fergal about it.